### PR TITLE
Add option to use yaml config data

### PR DESCRIPTION
**Pull request by lewismarshall**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -358,6 +362,25 @@ spec:
 Configuration can be provided via cli flags and arguments as well as
 environment variables.
 
+### .env
```
Comment (KashifSaadat) [inline README.md]: nitpick: Header should be `config` to match the arg

File: README.md [hunk 2]
```diff
@@ -391,6 +414,7 @@ GLOBAL OPTIONS:
    --kube-username USERNAME, -u USERNAME  kubernetes auth USERNAME [$KUBE_USERNAME, $PLUGIN_KUBE_USERNAME]
    --kube-password PASSWORD, -p PASSWORD  kubernetes auth PASSWORD [$KUBE_PASSWORD, $PLUGIN_KUBE_PASSWORD]
    --config value                         Env file location [$CONFIG_FILE, $PLUGIN_CONFIG_FILE]
+   --config-data value                    Config data e.g. --config-data=Chart=./Chart.yaml [$KD_CONFIG_DATA, $PLUGIN_KD_CONFIG_DATA]
```
Comment (KashifSaadat) [inline README.md]: Bit confused by the use as `--config-data=Chart=./Chart.yaml`. In the example further up it's just the filepath provided to the flag.

Comment (KashifSaadat) [reply README.md]: Scoped under `Chart` it seems when looking into the code below. Probably good to include an example for this in the README also, and demonstrate how it's referred to in the deployment files.

Comment (lewismarshall) [reply README.md]: So --config-data value has two forms:

1. Scope=file.yaml - data from file available at .Scope.rootkey.subkey
2. file.yaml - data from file available at .rootkey.subkey

I'll try and make the docs clearer...

File: main.go [hunk 3]
```diff
@@ -389,6 +396,93 @@ func run(c *cli.Context) error {
...
+	logInfo.Printf("Loaded config data from %s", f)
+	// Load yaml
+	if err := yaml.Unmarshal(b, &conf); err != nil {
+		return nil, err
+	}
+	// Update any values which DON't exist from environment
+	if mergeEnv && reflect.TypeOf(conf).Kind() == reflect.Map {
+		confTyped := conf.(map[interface{}]interface{})
+		for k, v := range EnvToMap() {
+			if _, ok := confTyped[v]; !ok {
```
Comment (marcinc) [inline main.go]: `confTyped[k]` ?

Comment (lewismarshall) [reply main.go]: An 'Anytype' var in golang - https://tour.golang.org/methods/14 so `confTyped` is a strongly typed var with conf.

Comment (lewismarshall) [reply main.go]: oh I see - mmm the tests pass - probably got k and v the wrong way round

File: test/TestConfigData/data-with-env.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,4 @@
+---
+data: {{ .data }}
+test-data-from-env: {{ .KD_TEST_DATA }}
+PATH: {{ .PATH }}
```
Comment (lewismarshall) [inline test/TestConfigData/data-with-env.yaml.tmpl]: @marcinc Nice one, good catch. I've fixed the bug and added a test to catch it too (don't replace data from environment with same name).

**Pull request merged**